chore: moved upload embeddings out of experimental

brianshen3 opened the pull request:
### Linked issue(s)
[KOL-7957](https://linear.app/kolena/issue/KOL-7957/move-upload-embeddings-out-of-the-experimental-section)
### What change does this PR introduce and why?
Move `upload_dataset_embeddings` out of _experimental
### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated (Going to do this separate PR/ticket)

brianshen3 committed:
moved upload embeddings out of experimental

brianshen3 committed:
redirect old upload emb to new location, warning on depreciation

nankolena commented on kolena/_experimental/search/embeddings.py:
```diff
@@ -11,131 +11,21 @@
-from kolena.errors import InputValidationError
-
-
-def upload_embeddings(key: str, embeddings: List[Tuple[str, np.ndarray]]) -> None:
```
This is the function to upload embeddings for workflows. Logically it does make sense to move this out of experimental, but then it should not be part of `/dataset` because it's not used for datasets. To avoid complexities I think it's fine to keep this here. What do you think?

brianshen3 replied:
Yes, lets keep it here for now. If needed I can fix it later.

nankolena replied:
Making sure we're on the same page here - is the plan to revert the change on `upload_embeddings`?

brianshen3 replied:
Yes, I will revert it.

nankolena replied:
This doesn't look reverted still, maybe consider something like these changes? https://github.com/kolenaIO/kolena/compare/brianshen/kol-7957-move-upload-embeddings-out-of-the-experimental-section...nan/kol-7957

brianshen3 replied:
Thanks for the branch reference. I followed that and updated the PR.

nankolena commented on kolena/_experimental/search/embeddings.py:
```diff
@@ -11,131 +11,21 @@
+from kolena.dataset.search import upload_dataset_embeddings as new_upload_dataset_embeddings
 
 
 @with_event(event_name=EventAPI.Event.UPLOAD_DATASET_EMBEDDINGS)
```
nit - should remove this, otherwise the event would be recorded twice because this calls the other `upload_dataset_embeddings`

brianshen3 replied:
Updated removing the duplicate event. Good catch.

nankolena commented on kolena/_experimental/search/embeddings.py:
```diff
@@ -11,131 +11,21 @@
 
+    .. deprecated:: 1.54.2
+        Use :func:`kolena.dataset.search.upload_dataset_embeddings` instead.
+
```
Nice job adding deprecation notice! A couple questions:
- Should this be 1.55.0 (the expected next release)?
- Its not rendered nicely, is there some syntax issue?
<img width="1167" alt="image" src="https://github.com/user-attachments/assets/d51252b9-8df1-4a28-a196-45213d6d0d70" />

brianshen3 replied:
Removed from experimental to avoid confusions. Updated to 1.55.0 as it's expected to be part of next release. Good catch.

nankolena reviewed (commented):
I think you are missing some changes to the docs index too because it's currently still shown under the experimental page. Probably need to [remove this block](https://github.com/kolenaIO/kolena/blob/ed91015a9fc6f17f6652819d2cced84ea9d30471/docs/reference/experimental/index.md?plain=1#L7-L10) and check if you need to add anything for the dataset page

<img width="1167" alt="image" src="https://github.com/user-attachments/assets/9bce96b3-c9d5-4d23-a793-6e4c458fa9c9" />

nankolena commented on kolena/_experimental/search/embeddings.py:
```diff
@@ -144,4 +34,4 @@ def upload_dataset_embeddings(dataset_name: str, key: str, df_embedding: pd.Data
     :raises InputValidationError: The provided input is not valid.
     """
-    _upload_dataset_embeddings(dataset_name, key, df_embedding)
+    new_upload_dataset_embeddings(dataset_name, key, df_embedding)
```
nit - consider adding a warning message here so that users will know this is deprecated.

brianshen3 replied:
Added deprecation warnings and use new endpoint

brianshen3 committed:
updated docs to reflect moved upload embeddings. Added deprecation warning

brianshen3 commented:
> I think you are missing some changes to the docs index too because it's currently still shown under the experimental page. Probably need to [remove this block](https://github.com/kolenaIO/kolena/blob/ed91015a9fc6f17f6652819d2cced84ea9d30471/docs/reference/experimental/index.md?plain=1#L7-L10) and check if you need to add anything for the dataset page
> 
> <img alt="image" width="1167" src="https://private-user-images.githubusercontent.com/145366880/411661983-9bce96b3-c9d5-4d23-a793-6e4c458fa9c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyMTUyOTMsIm5iZiI6MTczOTIxNDk5MywicGF0aCI6Ii8xNDUzNjY4ODAvNDExNjYxOTgzLTliY2U5NmIzLWM5ZDUtNGQyMy1hNzkzLTZlNGM0NThmYTljOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxMFQxOTE2MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NzkyMTc5NDY1MDI0MTliNmU2MGVlNjg0YmVjZDU1Njg3MmQyNTBjY2EyNWU3OGNkMjgwMzkwMmMyMjUwOWU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2XleIuP6kv_8S18cIkhStMrvqhxPC9_4R-NAvcYb2fY">

Updated to show properly under dataset. Removed showing from experimental to avoid confusion and follow current doc behaviour.

nankolena commented on docs/reference/dataset/index.md:
```diff
@@ -7,3 +7,7 @@
 ::: kolena.dataset.dataset
 ::: kolena.dataset.evaluation
+::: kolena.dataset.search
+    options:
+        show_root_heading: true
+        members: ["upload_dataset_embeddings"]
```
nit - let's remove these options to be consistent with others, currently this is the only method that's a level under

<img width="228" alt="image" src="https://github.com/user-attachments/assets/3edc23c9-f7cb-46a2-a03b-b8f2d06ba927" />

brianshen3 replied:
Fixed this

brianshen3 committed:
reverted changes to internal embedding. Fixed circular import issue.

brianshen3 committed:
updated integration test

brianshen3 committed:
updated circleCI to look at tests in dataset instead of experimental

brianshen3 committed:
reverted workflow embeddings. Cleaned up code

brianshen3 committed:
updated the testing to reverted embeding location.

brianshen3 committed:
moved embedding out of search

nankolena commented on tests/integration/dataset/search/test_embeddings.py:
```diff
@@ -18,8 +18,8 @@
 import pytest
```
nit: also bump this test file up similar to the structure in `kolena/dataset/embeddings.py`, and then we should be able to remove the `integration-test-dataset-search-embedding-<< matrix.python-version >>-<< matrix.extras >>` from the circleci config file because the following should take care of it: https://github.com/kolenaIO/kolena/blob/ed91015a9fc6f17f6652819d2cced84ea9d30471/.circleci/continue_config.yml#L268-L274

brianshen3 replied:
I removed the `integration-test-dataset-search-embedding-<< matrix.python-version >>-<< matrix.extras >>`. Seems to be a duplicate.

nankolena replied:
Could you also bump this integration test one folder up and get rid of the `/search` dir similar to how you did for `kolena/dataset/embeddings.py`?

nankolena commented on docs/reference/dataset/index.md:
```diff
@@ -7,3 +7,7 @@
 ::: kolena.dataset.dataset
 ::: kolena.dataset.evaluation
+::: kolena.dataset.embeddings
+    options:
+        members: ["upload_dataset_embeddings"]
+        show_root_heading: false
```
nit: I feel like these options are not needed here, could you verify whether removing them will not affect anything and if so get rid of these lines?

brianshen3 replied:
The reason I added the option was because the DatasetEmbeddingsDataFrameSchema shows up.
![Screenshot 2025-02-12 at 10 11 24 AM](https://github.com/user-attachments/assets/e0ffec97-eee0-4ee1-b9a1-99d6bd9957c5)

nankolena reviewed (commented):
Only a couple small nits - should be good once those are addressed!

brianshen3 committed:
removed duplicate integration test

brianshen3 committed:
moved integration test to match path in dataset api
